### PR TITLE
Use secrets for VCR based specs

### PR DESCRIPTION
**Pull request by Ladas**
Use secrets for VCR based specs

### CODE REVIEW

File: spec/models/manageiq/providers/amazon/aws_refresher_spec_common.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def expected_table_counts
       :floating_ip                   => 12,
       :guest_device                  => 0,
       :hardware                      => 46,
-      :miq_queue                     => 50,
```
Comment (durandom) [inline spec/models/manageiq/providers/amazon/aws_refresher_spec_common.rb]: huh? Why does this change?

Comment (Ladas) [inline spec/models/manageiq/providers/amazon/aws_refresher_spec_common.rb]: I was asking myself the same question :-)

**Comment by Ladas** [comment]
@durandom packed into Factory girl

File: spec/models/manageiq/providers/amazon/regions_spec.rb [hunk 2]
```diff
@@ -1,11 +1,7 @@
 describe ManageIQ::Providers::Amazon::Regions do
   it "has all the regions" do
-    EvmSpecHelper.create_guid_miq_server_zone
-    ems = FactoryGirl.create(:ems_amazon_with_authentication)
-    ems.update_authentication(:default => {
-                                :userid   => "0123456789ABCDEFGHIJ",
-                                :password => "ABCDEFGHIJKLMNO1234567890abcdefghijklmno"
-                              })
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
```
Comment (durandom) [inline spec/models/manageiq/providers/amazon/regions_spec.rb]: Do we always need a zone? Maybe we can move this into the factory too?

Comment (Ladas) [inline spec/models/manageiq/providers/amazon/regions_spec.rb]: not sure, I know it was failing somewhere deep without it

File: spec/factories/ext_management_system.rb [hunk 3]
```diff
@@ -5,4 +5,22 @@
           :parent  => :ems_storage do
     provider_region "us-east-1"
   end
+
+  factory :ems_amazon_with_vcr_authentication, :parent => :ems_amazon do
+    zone do
```
Comment (durandom) [inline spec/factories/ext_management_system.rb]: this is great!!!

File: spec/models/manageiq/providers/amazon/aws_refresher_spec_common.rb (hunk 1 above)
Comment (durandom) [inline spec/models/manageiq/providers/amazon/aws_refresher_spec_common.rb]: ok. can you dig some deeper? I dont think we should merge it without knowing why it changed... after all these numbers are specs!

**Comment by Ladas** [comment]
@durandom I would ignore the MiqQueue, several things can be writing there and that number changes a lot

**Comment by durandom** [comment]
Ok, so, actually the `miq_queue` numbers dont have anything to do with the tests..
Let's remove them ✂️

**Comment by durandom** [comment]
> Ok, so, actually the miq_queue numbers dont have anything to do with the tests..

@Ladas says he does this in a follow up

**Pull request merged**